tests/tx/scripts/test_sighash.py

```diff
@@ -0,0 +1,321 @@
+        htr_input.data = P2PKH.create_input_data(public_bytes2, signature2)
+
+        # The atomic swap tx is now completed and valid, and can be propagated
+        self.manager1.verification_service.verify(atomic_swap_tx_clone)
+        self.manager1.propagate_tx(atomic_swap_tx_clone, fails_silently=False)
+
+    @patch('hathor.transaction.scripts.opcode.is_opcode_valid', lambda _: True)
+    def test_sighash_bitmask_with_limit(self) -> None:
+        # Create a new test token
+        token_creation_tx = create_tokens(self.manager1, self.address1_b58)
+        token_uid = token_creation_tx.tokens[0]
+        token_creation_utxo = token_creation_tx.outputs[0]
+        genesis_utxo = self.genesis_block.outputs[0]
+        parents = self.manager1.get_new_tx_parents()
+
+        # Alice creates an input spending all created test tokens
+        tokens_input = TxInput(not_none(token_creation_tx.hash), 0, b'')
+
+        # Alice creates an output sending half genesis HTR to herself
+        alice_output_script = P2PKH.create_output_script(self.address1)
+        htr_output = TxOutput(int(genesis_utxo.value / 2), alice_output_script)
+
+        # Alice creates an atomic swap tx that's missing only Bob's input, with half genesis HTR
+        atomic_swap_tx = Transaction(
+            weight=1,
+            inputs=[tokens_input],
+            outputs=[htr_output],
+            parents=parents,
+            tokens=[token_uid],
+            storage=self.manager1.tx_storage,
+            timestamp=token_creation_tx.timestamp + 1
+        )
+        self.manager1.cpu_mining_service.resolve(atomic_swap_tx)
+
+        # Alice signs her input using sighash bitmasks, instead of sighash_all.
+        # She also sets max inputs and max outputs limits, including one output for change.
+        sighash_bitmask = SighashBitmask(inputs=0b1, outputs=0b1)
+        data_to_sign1 = atomic_swap_tx.get_custom_sighash_data(sighash_bitmask)
+        assert self.manager1.wallet
+        public_bytes1, signature1 = self.manager1.wallet.get_input_aux_data(data_to_sign1, self.private_key1)
+        tokens_input.data = P2PKH.create_input_data(
+            public_key_bytes=public_bytes1,
+            signature=signature1,
+            sighash=sighash_bitmask,
+            inputs_outputs_limit=InputsOutputsLimit(max_inputs=2, max_outputs=3)
+        )
+
+        # At this point, the tx is partial. The inputs are valid, but they're mismatched with outputs
+        self.manager1.verification_service.verifiers.tx.verify_inputs(atomic_swap_tx)
+        with pytest.raises(InputOutputMismatch):
+            self.manager1.verification_service.verify(atomic_swap_tx)
+
+        # Alice sends the tx bytes to Bob, represented here by cloning the tx
+        atomic_swap_tx_clone = cast(Transaction, atomic_swap_tx.clone())
+        self.manager1.cpu_mining_service.resolve(atomic_swap_tx_clone)
```

Comment:
Same here.

Reply:
The tx is verified on line 201, so it needs to be resolved